lib/itk/queue/consumer.ex

```diff
@@ -45,22 +57,72 @@ defmodule ITKQueue.Consumer do
     {:noreply, state}
   end
 
+  @doc false
+  def handle_info({_ref, _channel}, state) do
+    # extraneous message from AMQP.Channel.ReceiverManager
+    {:noreply, state}
+  end
+
+  @doc false
+  def handle_info({:DOWN, _ref, :process, _pid, reason}, state = %{subscription: sub}) do
+    Logger.error(
+      "Channel subscribed to #{sub.queue_name} (#{sub.routing_key}) went down: #{inspect(reason)}"
+    )
+
+    Process.send_after(self(), :subscribe, @reconnect_interval)
+    {:noreply, state}
+  end
+
+  @doc false
+  def handle_info(:subscribe, %__MODULE__{subscription: subscription}) do
+    state = subscribe(subscription)
+    {:noreply, state}
+  end
+
+  @doc false
+  def handle_call(:connection, _from, state) do
+    {:reply, connection(), state}
+  end
+
+  defp connection do
+    case GenServer.call(ITKQueue.ConsumerConnection, :connection) do
+      nil -> {:error, :connection_lost}
+      connection -> {:ok, connection}
+    end
+  catch
```

Comment:
I think you might have missed `try` in this function

Reply:
I changed that because credo said so, run `mix credo --all --strict`. Example here https://github.com/rrrene/credo/issues/224